### PR TITLE
Request history graphql

### PR DESCRIPTION
**Pull request by bluquar**
Screen Shot 2022-01-08 at 8 42 54 PM

### CODE REVIEW

File: src/server/objects/aid_request/AidRequestModel.ts [hunk 1]
```diff
@@ -1,12 +1,33 @@
...
+const AidRequestHistoryEventPayloadSchema =
+  new Schema<AidRequestHistoryEventPayload>({
+    event: {
+      enum: ['Completed', 'WorkingOn'],
+      type: String,
+    },
+  });
+
+const AidRequestHistoryEventSchema = new Schema<AidRequestHistoryEvent>({
+  action: { enum: ['Add', 'Remove'], type: String },
```
Comment (lowellbander) [inline src/server/objects/aid_request/AidRequestModel.ts:18]: why is there an `event` and an `action`? what's the difference?

File: src/server/objects/aid_request/AidRequestModelTypes.ts [hunk 2]
```diff
@@ -1,9 +1,40 @@
+import type { ObjectId } from 'mongodb';
+import { Schema } from 'mongoose';
+
+export type AidRequestHistoryEventPayload =
```
Comment (lowellbander) [inline src/server/objects/aid_request/AidRequestModelTypes.ts:4]: mongoose/compose doesn't generate half of this for u? seems weird to me that u basically have to write this out twice

File: src/server/objects/aid_request/AidRequestGraphQLTypes.ts [hunk 3]
```diff
@@ -7,3 +8,21 @@ export const AidRequestGraphQLType =
...
+    },
+    name: 'AidRequestHistoryEventPayload',
+  });
+
+export const AidRequestHistoryEventGraphQLType = schemaComposer.createObjectTC({
+  fields: {
+    action: 'String!',
+    actor: 'User',
+    details: 'AidRequestHistoryEventPayload',
+    timestamp: 'Date!',
```
Comment (lowellbander) [inline src/server/objects/aid_request/AidRequestGraphQLTypes.ts:25]: does `Date` counterintuitively include Time?

File: src/server/objects/aid_request/AidRequestModelTypes.ts [hunk 4]
```diff
@@ -1,9 +1,40 @@
...
+};
+
 export type AidRequestType = {
   completed: boolean;
   createdAt: Date;
+  history: AidRequestHistoryEvent[];
   whatIsNeeded: string;
   whoIsItFor: string;
-  whoIsWorkingOnIt: string[];
+  whoIsWorkingOnIt: ObjectId[];
```
Comment (lowellbander) [inline src/server/objects/aid_request/AidRequestModelTypes.ts:32]: why `ObjectId` rather than `User`?

**Review by lowellbander — APPROVED** [review]
this is very exciting!!!!

i don't understand why there's both action & event -- i guess specifically, what's this "Add/Remove" business?

also, not in scope for this PR, but `username` seems to actually be `email`, and we'll probs want actual `username` eventually so we're not rendering emails (probs block email from being exposed via GQL?)

**Comment by bluquar** [comment]
Yeah I thought about adding a comment to explain that but didn't :p Add/Remove is like Add="Marked themself as working on it"/"Marked the task as Complete", and Remove="Removed themself as working on it"/"Removed Completed status"

**Comment by lowellbander** [comment]
> Yeah I thought about adding a comment to explain that but didn't :p Add/Remove is like Add="Marked themself as working on it"/"Marked the task as Complete", and Remove="Removed themself as working on it"/"Removed Completed status"

oh ok! ty for explain!

File: src/server/objects/aid_request/AidRequestModelTypes.ts (hunk 2 above)
Comment (bluquar) [reply src/server/objects/aid_request/AidRequestModelTypes.ts:4]: no :( at least I couldn't figure out how to get it to follow object references

File: src/server/objects/aid_request/AidRequestModelTypes.ts (hunk 4 above)
Comment (bluquar) [reply src/server/objects/aid_request/AidRequestModelTypes.ts:32]: this is for what's stored in the DB, which is the ID of the user that's being referenced

**Pull request merged**